lib/doorkeeper/config.rb

```diff
@@ -228,33 +228,24 @@ def realm
       @realm ||= 'Doorkeeper'
     end
 
-    def authorization_response_types
-      @authorization_response_types ||= calculate_authorization_response_types
+    def enabled_grant_flows
+      @enabled_grant_flows ||= grant_flows.map{|name| GrantFlow.get(name) }
     end
 
-    def token_grant_types
-      @token_grant_types ||= calculate_token_grant_types
+    def authorization_response_flows
+      @authorization_response_types ||= enabled_grant_flows.select(&:handles_response_type?)
```

Comment:
I feel like this ends up being more readable than splitting the select across multiple lines.